pkg/api/validation/validation.go

```diff
@@ -1147,6 +1148,9 @@ func validateServicePort(sp *api.ServicePort, requireName bool, allNames *util.S
 		if sp.TargetPort.Kind == util.IntstrInt && !util.IsValidPortNum(sp.TargetPort.IntVal) {
 			allErrs = append(allErrs, errs.NewFieldInvalid("targetPort", sp.TargetPort, portRangeErrorMsg))
 		}
+		if sp.TargetPort.Kind == util.IntstrString && !util.IsValidPortString(sp.TargetPort.StrVal) {
```

Comment:
Please also drop the test for = 0 or "" - I assert they are not needed any more (since v1beta1 was removed)